x-pack/plugins/rule_registry/server/alert_data_client/alerts_client.ts

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { PublicMethodsOf } from '@kbn/utility-types';
+import { AlertTypeParams } from '../../../alerting/server';
+import {
+  ReadOperations,
+  AlertingAuthorization,
+  WriteOperations,
+  AlertingAuthorizationEntity,
+  // eslint-disable-next-line @kbn/eslint/no-restricted-paths
+} from '../../../alerting/server/authorization';
+import { Logger, ElasticsearchClient } from '../../../../../src/core/server';
+import { alertAuditEvent, AlertAuditAction } from './audit_events';
+import { RuleDataPluginService } from '../rule_data_plugin_service';
+import { AuditLogger } from '../../../security/server';
+import { OWNER, RULE_ID } from '../../common/technical_rule_data_field_names';
+import { ParsedTechnicalFields } from '../../common/parse_technical_fields';
+
+export interface ConstructorOptions {
+  logger: Logger;
+  authorization: PublicMethodsOf<AlertingAuthorization>;
+  auditLogger?: AuditLogger;
+  esClient: ElasticsearchClient;
+  ruleDataService: PublicMethodsOf<RuleDataPluginService>;
+}
+
+export interface UpdateOptions<Params extends AlertTypeParams> {
+  id: string;
+  data: {
+    status: string;
+  };
+  // observability-apm see here: x-pack/plugins/apm/server/plugin.ts:191
+  indexName: string;
+}
+
+interface GetAlertParams {
+  id: string;
+  // observability-apm see here: x-pack/plugins/apm/server/plugin.ts:191
+  indexName: string;
+}
+
+export class AlertsClient {
+  private readonly logger: Logger;
+  private readonly auditLogger?: AuditLogger;
+  private readonly authorization: PublicMethodsOf<AlertingAuthorization>;
+  private readonly esClient: ElasticsearchClient;
+  private readonly ruleDataService: PublicMethodsOf<RuleDataPluginService>;
+
+  constructor({
+    auditLogger,
+    authorization,
+    logger,
+    esClient,
+    ruleDataService,
+  }: ConstructorOptions) {
+    this.logger = logger;
+    this.authorization = authorization;
+    this.esClient = esClient;
+    this.auditLogger = auditLogger;
+    this.ruleDataService = ruleDataService;
+  }
+
+  public getFullAssetName() {
+    return this.ruleDataService?.getFullAssetName();
+  }
+
+  public async getAlertsIndex(featureIds: string[]) {
+    return this.authorization.getAuthorizedAlertsIndices(
+      featureIds.length !== 0 ? featureIds : ['apm', 'siem']
```

Comment:
Again I'm not following why APM and SIEM are hard-coded here.

Reply:
https://github.com/elastic/kibana/pull/100705#discussion_r655696777